### DISCUSSION TITLE
Is there a way to feed random property of Record?

### DISCUSSION THREAD

**Discussion by GuillaumeUnice** [dubzzz/fast-check#3641] [Q&A] [answered]
Hi every,
I would like to also randomize my field of the record but keep it to then perform my property. So something like that:

```
 const a = fc.string();
const b = fc.string();
const c = fc.string();
const r = fc.record({
  [a]: fc.constant(undefined),
  [b]: fc.oneof(fc.string(), fc.boolean(), fc.integer(), fc.float()),
  [c]: fc.constant(null),
})
// Here naturally the a, b & c or yet Arbitrary thats my problem cause I want to keep reference of them in my property as follow
const foo = fc.property(a, b, c, r(a, b, c), (a, b, c, rs) => {
  const generatedId = generateId(rs, [a, b, c]);
  return generatedId === b;
});
fc.assert(foo, { numRuns: 10000 });
```

Hope the request is clear, don't hesitate in other case ;)

**Comment by dubzzz** [comment, marked as answer]
You can use an `fc.tuple` or `fc.record` to generate a, b and c and then use a `.chain` returning a, b, c and the new value.

In one of the upcoming releases, I'll add another util that should help users into drafting code with such patterns.

**Reply by dubzzz** [reply]
You may see it as:

```js
const a = fc.string();
const b = fc.string();
const c = fc.string();
const all = fc.record({a,b,c})
  .chain(data => fc.record({
    data: fc.constant(data),
    other: fc.record({
      [data.a]: fc.constant(undefined),
      [data.b]: fc.oneof(fc.string(), fc.boolean(), fc.integer(), fc.float()),
      [data.c]: fc.constant(null),
    })
  })
```

The record of record in chain is just to avoid the risk of a, b, c from being equal to data.

**Reply by GuillaumeUnice** [reply]
Merci!

I dont really grab the reason "being equal to data" but I got the process by "lazing" so thanks and keeping reference to a,b,c is helpful for the property test.
